app/users/user.service.ts

```diff
@@ -225,20 +255,35 @@ export class UserService {
    * @returns {Promise<string>}
    */
   public updateStudentDetails(user: User, isCurrent = false) {
-    return this.cmsApiService.putUrl('/Api/User/Student' + (!isCurrent ? '?identity=' + user.email : ''),
-                                      JSON.stringify(user.studentDetails), {})
+    return this.userStoreApiService.putUrl('/api/Users/' + (!isCurrent ? '?id=' + user.id : '') + '/StudentDetails',
+      JSON.stringify(user.studentDetails), {})
       .toPromise()
       .then(
-        (response: Response) => {
-          if (response.status === 200) {
-            return 'Information successfully updated';
-          }
+      (response: Response) => {
+        if (response.status === 200) {
+          return 'Information successfully updated';
         }
+      }
       ).catch(
-        (error: any) => this.handleError(error)
+      (error: any) => this.handleError(error)
       );
   }
 
+  /**
+   * Delete the student details for the given user.
+   * @param isCurrent updating the current user? Default value is false.
+   * @returns {Promise<string>}
+   */
+  public deleteStudentDetails(id: string, isCurrent = false) {
+    return this.userStoreApiService.deleteUrl('api/Users/' + (!isCurrent ? '?id=' + id : '') + '/StudentDetails', {})
```

Comment:
This looks like it might produce URL like `api/Users/?id=auth0|blabla/StudentDetails` which would be wrong. It should be `api/Users/auth0|blabla/StudentDetails`